sdks/java/core/src/main/java/org/apache/beam/sdk/options/ProxyInvocationHandler.java

```diff
@@ -142,6 +165,41 @@ public Object invoke(Object proxy, Method method, Object[] args) {
   }
 
   /**
+   * Track whether options values are explicitly set, or retrieved from deserialized JSON/defaults.
```

Comment:
it won't be retrieved from deserialized JSON since that is contained completely within a different map.

This just tracks whether came from a default vs being explicitly set.